packages/python/plotly/plotly/express/_core.py

```diff
@@ -1419,9 +1419,17 @@ def build_dataframe(args, constructor):
             else:
                 # Save precious resources by only interchanging columns that are
                 # actually going to be plotted.
-                columns = [
+                necessary_columns = [
                     i for i in args.values() if isinstance(i, str) and i in columns
                 ]
+                for field in args:
+                    if field in array_attrables and isinstance(
+                        args[field], (list, dict)
+                    ):
+                        necessary_columns.extend(
+                            [i for i in args[field] if i in columns]
+                        )
+                columns = list(dict.fromkeys(necessary_columns))
```

Comment:
Is this just to limit to unique items? Maybe clearer as `list(set(necessary_columns))` then?

Reply:
I used it to preserve the order of the columns, but will change to your suggestion as the order doesn't seem to be important here

Reply:
ha ok - the order there doesn't actually matter, so that test could be rewritten to turn the assertion into a set or sorted list comparison `assert sorted(interchange_dataframe.select_columns_by_name.call_args.args[0]) == ["petal_width", "sepal_length"]` or some such. But not a big deal.

Reply:
Got it! thanks. (btw I tried using `set()` but vaex requires the columns passed to `select_columns_by_name()` to be a `Sequence` for some reason, so I ended up using your suggestion)

Reply:
Oh, this `call_args.args` property was introduced in python 3.8, so I'm slightly changing it